libraries/DataFlash/LogFile.cpp

```diff
@@ -1437,6 +1437,14 @@ void DataFlash_Class::Log_Write_Current_instance(const uint64_t time_us,
 // Write an Current data packet
 void DataFlash_Class::Log_Write_Current()
 {
+    // Big painful assert to ensure that logging won't produce suprising results when the
+    // number of battery monitors changes
+    static_assert(AP_BATT_MONITOR_MAX_INSTANCES == 9,
```

Comment:
I'm not sure how this fixes much, just changing here to the correct number makes this pass. I think a better check would be `LOG_CURRENT_MSG + AP_BATT_MONITOR_MAX_INSTANCES == LOG_CURRENT_CELLS_MSG`. It still doesn't make sure that cells enum entries are updated, but I haven't come up with a way for it.